src/librustc_typeck/diagnostics.rs

```diff
@@ -380,6 +380,52 @@ fn main() {
 ```
 "##,
 
+E0044: r##"
+You can't use type parameters on foreign items. Example of erroneous code:
+
+```
+extern { fn some_func<T>(); }
+```
+
+Just remove the type parameter to make this code works:
```

Comment:
Well, the current example would likely break any code using generics. You should change the advice to not recommend "just removing the type parameter".

Something like "to fix this, replace the type parameter with the specializations that you need"

Examples would be:

```
extern {
  fn some_func_i32(x: i32); 
  fn some_func_f64(x: f64);
}
```

over `fn some_func<T>(x: T)`